crates/bevy_ecs/src/hierarchy.rs

```diff
@@ -185,6 +185,33 @@ impl<'w> EntityWorldMut<'w> {
         self.add_related::<ChildOf>(children)
     }
 
+    /// Insert children at specific index
+    pub fn insert_children(&mut self, index: usize, children: &[Entity]) -> &mut Self {
+        let parent = self.id();
+        if children.contains(&parent) {
+            panic!("Cannot insert entity as a child of itself.");
```

Comment:
Is it possible to change the return to Bevy Result and then return `Err()` here?